docs/cli.md

```diff
@@ -114,9 +114,54 @@ In order to store and reuse commonly used CLI options, you can add a `cucumber.j
 Use `--tags <EXPRESSION>` to run specific features or scenarios. This option is repeatable and the expressions will be merged with an `and` operator.
 `<EXPRESSION>` is a [cucumber tag expression](https://docs.cucumber.io/cucumber/api/#tag-expressions).
 
-## Transpilers
+## Transpilation
 
-Step definitions and support files can be written in other languages that transpile to JavaScript. To do this use the CLI option `--require-module <module_name>`, where `require("<module_name>")` should make it possible to require files for your language. Also use `--require 'features/**/*.<ext>'` if your files end in an extension other than `js`. As an example, load [CoffeeScript](https://www.npmjs.com/package/coffee-script) support files with `--require-module coffee-script/register --require 'features/**/*.coffee'`.
+Step definitions and support files can be written in other languages that transpile to JavaScript.
+
+### Simple ES6 support
+
+For instance, for ES6 support with [Babel](https://babeljs.io/) 7 add:
+
+```
+--require-module @babel/register
+```
+
+This will effectivally call `require('@babel/register')` prior to requiring any support files.
+
+### Non JS files
+
+If your files end in an extension other than `js`, make sure to also include the `--require` option to state the support files to require.
+
+For example, with [TypeScript](https://www.typescriptlang.org/):
+
+```
+--require-module ts-node/register --require step-definitions/**/*.ts
+```
+
+or [CoffeeScript](https://www.npmjs.com/package/coffee-script):
+
+```
+--require-module coffee-script/register --require 'features/**/*.coffee'
+```
+
+### Extra configuration
+
+Sometimes the required module (say `@babel/register`) needs extra configuration. In such cases create a script (say, `cucumber.config.js`):
```

Comment:
Thoughts on the sample file being `babel_register.js`? Its not cucumber specific and could actually be used if you have mocha unit tests.

Reply:
Naming things!

I completely see where you are coming from, but this is SOOOOOO context dependent!

One things we have done in the past is `chai.use(chaiAsPromised);` in such script. But I guess that's for Mocha as well (but `babel` in the name won't make sense).

Possibly `tests.setup.js` is another candidate... doubt anything will come out as an ultimate winner here...

Anyhow, changed to `babel.register.js`.

(Don't mind changing to `tests.setup.js` though).

Views?

Reply:
Either `babel.register.js` or `tests.setup.js` work for me.